### DISCUSSION TITLE
Cloudflare Worker + Remix + Clerk Starter

### DISCUSSION THREAD

**Discussion by aerojeyenth** [clerk/javascript#1967] [Feature Requests]
## Package

- [x] `@clerk/remix`

## Description

The most important USP of Clerk in my humble opinion is its capability to work on `Edge `like Cloudflare Workers. Please kindly provide a getting started tutorial and a sample starter for using Clerk in `Cloudflare Workers with Remix`.

**Comment by Ehesp** [comment]
You just need to follow https://clerk.com/docs/quickstarts/get-started-with-remix and deploy to Cloudflare: https://remix.run/docs/en/1.19.1/guides/envvars#production

**Comment by aerojeyenth** [comment]
@Ehesp Thank you so much for your kind response. 
But I would like to tell you I tried the docs before and it never worked on Cloudflare Workers!! Lot of work arounds are needed before we can make it work. For example recently CF workers stopped exposing the ENV variables in the globalThis scope and breaks the Clerk's implementation. (ref: [https://github.com/clerkinc/javascript/issues/1540](https://github.com/clerkinc/javascript/issues/1540))

This is not just for the benefit of me and for a wider audience it be of great help and prevent the huge friction that happens while trying to adapt Clerk.

**Comment by JacobMGEvans** [comment]
Hmmm, I could try putting a very simple example together. I might not be able to get to it until this weekend.

**Comment by JacobMGEvans** [comment]
Ah! I forgot about this, you can blame Baldurs Gate 3 

I will dedicate some time tomorrow morning to this.

**Comment by JacobMGEvans** [comment]
Ok update, I talked to with someone from Cloudflare Pages to see if there was any potential better ways that this can be done. Overall the approach I will be taking is this:
 
1. Start with https://clerk.com/docs/quickstarts/get-started-with-remix
2. Implement workaround in https://github.com/clerkinc/javascript/issues/1540
3. See if that works
4. Work on trying to find a better way to deal with auth, specifically with CF Workers

At the end I hope to have a very simple concise example (a critical success would be agnostic even to Remix)

**Comment by LekoArts** [comment]
Hello!

We're tracking our feature requests now in [feedback.clerk.com](https://feedback.clerk.com). If your suggestion is already in the list of proposed changes, please upvote. Otherwise, please create an entry. Thanks!

Our feedback platform will be the source of truth for feature requests going forward and your input and votes will help us shape the roadmap.

Thanks for using Clerk! 💙

**Comment by bramses** [comment]
Does this work with any CF Worker setup, or does it have to use Remix? I'm using the hono framework (https://hono.dev/), but would like to attempt to protect some routes with Clerk

**Reply by panteliselef** [reply]
Does this help? 

https://github.com/honojs/middleware/tree/main/packages/clerk-auth

**Reply by bramses** [reply]
Oh awesome! Thank you @panteliselef!
